dspace-api/src/main/java/org/dspace/app/mediafilter/ImageMagickThumbnailFilter.java

```diff
@@ -132,12 +137,19 @@ public static File getImageFile(File f, int page) throws IOException, Interrupte
     	f2.deleteOnExit();
     	ConvertCmd cmd = new ConvertCmd();
 		IMOperation op = new IMOperation();
+		Info imageInfo = new Info(f.getAbsolutePath(),true);
 		String s = "[" + page + "]";
 		op.addImage(f.getAbsolutePath()+s);
                 if (flatten)
                 {
                     op.flatten();
                 }
+		String imageClass = imageInfo.getImageClass();
+		// PDFs using the CMYK color system can be handled specially if profiles are defined
+		if (imageClass.contains("CMYK") && cmyk_profile != null && srgb_profile != null) {
+			op.profile(cmyk_profile);
+			op.profile(srgb_profile);
```

Comment:
In imagemagick's documentation is written: "If there is no embedded profile then the first "-profile" is the input profile. A second "-profile" then defines the output profile. If there is an embedded profile then a single of "-profile" operator will immediately define the output profile" (see http://www.imagemagick.org/Usage/formats/#color_profile_changing). Shouldn't we check if a color profile is embedded within the PDF before assigning two profiles? While the output profile should be always fine, especially the input profile should match the source's profile as close as possible.